core/web/config_controller.go

```diff
@@ -32,7 +32,10 @@ func (cc *ConfigController) Dump(c *gin.Context) {
 		jsonAPIError(c, http.StatusInternalServerError, err)
 		return
 	}
-	jsonAPIResponse(c, tomlStr, "config")
+
+	// Return a human-readable TOML config. It's not a valid JSON so
+	// technically it violates jsonAPI assumptions.
+	tomlResponse(c, tomlStr)
```

Comment:
Can you elaborate on the problem here? My intent was to return a TOML string inside of a JSON field. Is there some reason that cannot work?

Reply:
That could work too (but was broken anyway). I thought it would be more convenient if we return plaintext TOML, so that it can be saved straight to a file, instead of requiring an additional step of parsing a JSON wrapper. If we prefer to always return JSON I can change it.

Reply:
FYI this is the error I'm getting from current code when hitting /v2/config/v2 path:
{"errors":[{"detail":"failed to marshal config using jsonapi: Marshal only accepts slice, struct or ptr types"}]}

Reply:
> so that it can be saved straight to a file

What do you mean?

> If we prefer to always return JSON I can change it.

Yes. This is a JSON API, and many things expect to interact with it by speaking only JSON, plus there are standard forms for errors, etc. - Adding TOML would only complicate things unecesssarily IMHO. There is no need to expose that detail since this is just a string for user presentation that shouldn't be interpretted at the API level anyways.

Reply:
I imagined a typical use-case for this command would be dumping config to save it and use for another node or possibly eyeball it to check some field values or compare to another config file. That will be much harder if we return JSON with no new lines etc. If that's not the case, what is the desired use case?

I can make it a valid JSON, no problem. Would something like {"config": "<actual_config_here>"} work? It has to be wrapped in a minimal dictionary or array + it needs to have escaped newlines etc.

Reply:
The CLI is meant to output plain TOML to the user, but the CLI is just one consumer of the JSON API.

No modifications of the string should be necessary - any escaping would happen automatically.

Something like `"config": "<toml>"` is what I expected to happen, but it looks like it doesn't work that way and insists on wrapped type implementing an interface.

If you want to fix this you can, but it is not in use yet and we're still working on the related pieces, so feel free to create a shortcut story instead: https://app.shortcut.com/chainlinklabs/epic/33587

Reply:
Ah, I see, that makes sense now, thanks :) I'll try to fix it. I already opened a story too: https://app.shortcut.com/chainlinklabs/story/46646/config-dump-command-is-broken